src/main/java/ru/vk/itmo/test/reference/ReferenceServer.java

```diff
@@ -90,6 +104,39 @@ public Response entity(Request request, @Param(value = "id") String id) {
             return new Response(Response.BAD_REQUEST, Response.EMPTY);
         }
 
+        String executorNode = getNodeByEntityId(id);
+
+        if (executorNode.equals(config.selfUrl())) {
+            return invokeLocal(request, id);
+        } else {
+            try {
+                return invokeRemote(executorNode, request);
+            } catch (IOException e) {
+                log.info("I/O exception while calling remote node");
+                return new Response(Response.INTERNAL_ERROR, Response.EMPTY);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                log.info("Thread interrupted");
+                return new Response(Response.SERVICE_UNAVAILABLE, Response.EMPTY);
+            }
+        }
+    }
+
+    private Response invokeRemote(String executorNode, Request request) throws IOException, InterruptedException {
+        HttpRequest httpRequest = HttpRequest.newBuilder(URI.create(executorNode + request.getURI()))
+            .method(
+                request.getMethodName(),
+                request.getBody() == null
+                    ? HttpRequest.BodyPublishers.noBody()
+                    : HttpRequest.BodyPublishers.ofByteArray(request.getBody())
+            )
+            .timeout(Duration.of(500, ChronoUnit.MILLIS))
```

Comment:
Можно чуточку лучше:
```suggestion
            .timeout(Duration.ofMillis(500))
```